src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -81,7 +81,7 @@ public async Task<LocationOrLocationLinks> Handle(DefinitionParams request, Canc
                 return null;
             }
 
-            var originTagDescriptor = originTagHelperBinding.Descriptors.SingleOrDefault();
+            var originTagDescriptor = originTagHelperBinding.Descriptors.FirstOrDefault();
```

Comment:
Additional `Descriptors` would be the attributes, hence why this was failing earlier.

Reply:
lmao, yup and directive attributes would absolutely be responsible for that. That being said, we're not guaranteed for the correct TagHelper to be the "first" one now. We need to make sure it doesn't accidentally grab the directive attribute. To do that I'd imagine we'd need to look at all the TagHelpers and filter out inapplicable ones. Or is that done elsewhere?

Reply:
> we're not guaranteed for the correct TagHelper to be the "first" one now 

Hmm okay, wasn't sure if it was guaranteed, but wasn't getting any attributes ahead of the tag in my testing. Added the guard regardless to be safe. 

> Or is that done elsewhere?

We get these descriptors from `tagHelperElement.TagHelperInfo.BindingResult`.

Reply:
Ya binding result will also have the directive attributes so adding the filter is a good safeguard